File: src/nfan/fan/sys/Service.cs
//
// Copyright (c) 2009, Brian Frank and Andy Frank
// Licensed under the Academic Free License version 3.0
//
// History:
//   30 Mar 09  Andy Frank  Creation
//

using System;
using Fanx.Fcode;

namespace Fan.Sys
{
  /// <summary>
  /// Service.
  /// </summary>
  public interface Service
  {

    Service install();

    Service uninstall();

  }
}

File: src/jfan/fan/sys/Service.java
//
// Copyright (c) 2009, Brian Frank and Andy Frank
// Licensed under the Academic Free License version 3.0
//
// History:
//   26 Mar 09  Brian Frank  Creation
//
package fan.sys;

/**
 * Service
 */
public interface Service
{
  public boolean isInstalled();

  public boolean isRunning();

  public Service install();

  public Service uninstall();

  public Service start();

  public Service stop();

  public void onStart();

  public void onStop();

}